ticdc: add metrics doc (#4711)
pingcap/docs: ticdc/monitor-ticdc.md

![TiCDC Dashboard - Server metrics](/media/ticdc/ticdc-dashboard-server.png)

The description of each metric in the **Server** panel is as follows:

- Uptime: The time for which TiKV nodes and TiCDC nodes have been running
- Goroutine count: The number of goroutines of a TiCDC node
- Open FD count: The number of file handles opened by TiCDC nodes
- Ownership: The current status of nodes in the TiCDC cluster
- Ownership history: The ownership history of the TiCDC cluster
- CPU usage: The CPU usage of TiCDC nodes
- Memory usage: The memory usage of TiCDC nodes

## Changefeed

The following is an example of the **Changefeed** panel:

![TiCDC Dashboard - Changefeed metrics 1](/media/ticdc/ticdc-dashboard-changefeed-1.png)
![TiCDC Dashboard - Changefeed metrics 2](/media/ticdc/ticdc-dashboard-changefeed-2.png)
![TiCDC Dashboard - Changefeed metrics 3](/media/ticdc/ticdc-dashboard-changefeed-3.png)

The description of each metric in the **Changefeed** panel is as follows:

- Changefeed table count: The number of tables that each TiCDC node needs to replicate in the replication task
- Processor resolved ts: The timestamps that have been resolved in the TiCDC cluster

> - Table resolved ts: The replication process of each table in the replication task

Reviewer: ```suggestion
- Table resolved ts: The replication progress of each table in the replication task
```

- Changefeed checkpoint: The process of replicating data to the downstream. Normally, the green bars are connected to the yellow line
- PD etcd requests/s: The number of requests that a TiCDC node sends to PD per second
- Exit error count: The number of errors that interrupt the replication task per minute

> - Changefeed checkpoint lag: The data replication process lag (the unit is second) between the upstream and the downstream

Reviewer: ```suggestion
- Changefeed checkpoint lag: The process lag of data replication (the unit is second) between the upstream and the downstream
```

> - Changefeed checkpoint lag: The process lag of data replication (the unit is second) between the upstream and the downstream

Reviewer: ```suggestion
- Changefeed checkpoint lag: The progress lag of data replication (the unit is second) between the upstream and the downstream
```

> - Changefeed resolved ts lag: The data replication process lag (the unit is second) between the upstream and TiCDC nodes

Reviewer: ```suggestion
- Changefeed resolved ts lag: The process lag of data replication (the unit is second) between the upstream and TiCDC nodes
```

> - Changefeed resolved ts lag: The process lag of data replication (the unit is second) between the upstream and TiCDC nodes

Reviewer: ```suggestion
- Changefeed resolved ts lag: The progress lag of data replication (the unit is second) between the upstream and TiCDC nodes
```

- Flush sink duration: The histogram of the time spent by TiCDC asynchronously flushing data to the downstream

> - Flush sink duration percentile: Within one second, the 95%, 99%, and 99.9% probability of the time spent by TiCDC asynchronously flushing data to the downstream

Reviewer: ```suggestion
- Flush sink duration percentile: The time (P95, P99, and P999) spent by TiCDC asynchronously flushing data to the downstream within one second
```

- Sink write duration: The histogram of the time spent by TiCDC writing a transaction change to the downstream

> - Sink write duration percentile: Within one second, the 95%, 99%, and 99.9% probability of the time spent by TiCDC writing a transaction change to the downstream

Reviewer: ```suggestion
- Sink write duration percentile: The time (P95, P99, and P999) spent by TiCDC writing a transaction change to the downstream within one second
```

> - MySQL sink conflict detect duration: The histogram of the time spent by detecting MySQL sink conflicts

Reviewer: ```suggestion
- MySQL sink conflict detect duration: The histogram of the time spent on detecting MySQL sink conflicts
```

> - MySQL sink conflict detect duration percentile: Within one second, the 95%, 99%, and 99.9% probability of the time spent by detecting MySQL sink conflicts

Reviewer: ```suggestion
- MySQL sink conflict detect duration percentile: The time (P95, P99, and P999) spent on detecting MySQL sink conflicts within one second
```

- MySQL sink worker load: The workload of MySQL sink workers of TiCDC nodes

## Events

> Following is an example of the **Events** panel:

Reviewer: ```suggestion
The following is an example of the **Events** panel:
```

![TiCDC Dashboard - Events metrics 2](/media/ticdc/ticdc-dashboard-events-1.png)
![TiCDC Dashboard - Events metrics 2](/media/ticdc/ticdc-dashboard-events-2.png)
![TiCDC Dashboard - Events metrics 2](/media/ticdc/ticdc-dashboard-events-3.png)

The description of each metric in the **Events** panel is as follows:

> - Eventfeed count: TiCDC 节点中 Eventfeed RPC 的个数 The number of Eventfeed RPC of TiCDC nodes

Reviewer: ```suggestion
- Eventfeed count: The number of Eventfeed RPC requests of TiCDC nodes
```

> - Event size percentile: Within one second, the 95%, 99%, and 99.9% probability of the event size which TiCDC receives from TiKV

Reviewer: ```suggestion
- Event size percentile: The event size (P95, P99, and P999) which TiCDC receives from TiKV within one second
```

> - Eventfeed error/m: The number of errors reported by Eventfeed RPC of TiCDC nodes per minute

Reviewer: ```suggestion
- Eventfeed error/m: The number of errors reported by Eventfeed RPC requests of TiCDC nodes per minute
```

- KV client receive events/s: The number of events that the KV client module of TiCDC nodes receives from TiKV per second
- Puller receive events/s: The number of events that the Puller module of TiCDC nodes receives from the KV client per second
- Puller output events/s: The number of events that the Puller module of TiCDC nodes sends to the Sorter module per second
- Sink flush rows/s: The number of events that TiCDC nodes write to the downstream per second
- Puller buffer size: The number of events that TiCDC nodes cache in the Puller module
- Entry sorter buffer size: The number of events that TiCDC nodes cache in the Sorter module
- Processor/Mounter buffer size: The number of events that TiCDC nodes cache in the Processor module and the Mounter module
- Sink row buffer size: The number of events that TiCDC nodes cache in the Sink module

> - Entry sorter sort duration: TiCDC The histogram of the time spent by TiCDC nodes sorting events

Reviewer: ```suggestion
- Entry sorter sort duration: The histogram of the time spent by TiCDC nodes sorting events
```

> - Entry sorter sort duration percentile: Within one second, the 95%, 99%, and 99.9% probability of the time spent by TiCDC sorting events

Reviewer: ```suggestion
- Entry sorter sort duration percentile: The time (P95, P99, and P999) spent by TiCDC sorting events within one second
```

- Entry sorter merge duration: The histogram of the time spent by TiCDC nodes merging sorted events

> - Entry sorter merge duration percentile: Within one second, the 95%, 99%, and 99.9% probability of the time spent by TiCDC merging sorted events

Reviewer: ```suggestion
- Entry sorter merge duration percentile: The time (P95, P99, and P999) spent by TiCDC merging sorted events within one second
```

- Mounter unmarshal duration: The histogram of the time spent by TiCDC nodes unmarshaling events

> - Mounter unmarshal duration percentile: Within one second, the 95%, 99%, and 99.9% probability of the time spent by TiCDC unmarshaling events

Reviewer: ```suggestion
- Mounter unmarshal duration percentile: The time (P95, P99, and P999) spent by TiCDC unmarshaling events within one second
```

- KV client dispatch events/s: The number of events that the KV client module dispatches among the TiCDC nodes
- KV client batch resolved size: The batch size of resolved timestamp messages that TiKV sends to TiCDC

## TiKV

> Following is an example of the **TiKV** panel:

Reviewer: ```suggestion
The following is an example of the **TiKV** panel:
```

![TiCDC Dashboard - TiKV metrics 1](/media/ticdc/ticdc-dashboard-tikv-1.png)
![TiCDC Dashboard - TiKV metrics 2](/media/ticdc/ticdc-dashboard-tikv-2.png)

The description of each metric in the **TiKV** panel is as follows:

> - CDC endpoint CPU: The CPU usage of the CDC endpoint threads of TiKV nodes
> - CDC worker CPU: The CPU usage of the CDC worker threads of TiKV nodes

Reviewer: ```suggestion
- CDC endpoint CPU: The CPU usage of the CDC endpoint threads on TiKV nodes
- CDC worker CPU: The CPU usage of the CDC worker threads on TiKV nodes
```

- Min resolved ts: The minimum resolved timestamp on TiKV nodes

> - Min resolved region: The minimum Region ID of resolved timestamps on TiKV nodes

Reviewer: ```suggestion
- Min resolved region: The Region ID of the minimum resolved timestamps on TiKV nodes
```

> - Min resolved region: The minimum region ID of resolved timestamp on TiKV nodes

Reviewer: ```suggestion
- Min resolved region: The minimum Region ID of resolved timestamps on TiKV nodes
```

- Resolved ts lag duration percentile: The lag between the minimum resolved timestamp on TiKV nodes and the current time

> - Initial scan duration: The histogram of the time spent by incremental scan when TiKV nodes connect to TiCDC nodes

Reviewer: ```suggestion
- Initial scan duration: The histogram of the time spent on incremental scan when TiKV nodes connect to TiCDC nodes
```

> - Initial scan duration percentile: Within one second, the 95%, 99%, and 99.9% probability of the time spent by the incremental scan of TiKV nodes

Reviewer: ```suggestion
- Initial scan duration percentile: The time (P95, P99, and P999) spent on the incremental scan of TiKV nodes within one second
```

> - Memory without block cache: The memory usage of TiKV nodes, not including the RocksDB block cache

Reviewer: ```suggestion
- Memory without block cache: The memory usage of TiKV nodes excluding the RocksDB block cache
```
